Lib/test/test_zipfile/_path/test_path.py

```diff
@@ -665,7 +666,9 @@ def for_name(cls, name, archive):
 
         TODO: extract this functionality and re-use
         """
-        self = cls(filename=name, date_time=time.localtime(time.time())[:6])
+        epoch = int(os.environ.get('SOURCE_DATE_EPOCH')
+        get_time = int(epoch) if epoch is not None else time.time()
+        self = cls(filename=name, date_time=time.gmtime(get_time)[:6])
```

Comment:
Note that this duplication is undesirable and there's an effort underway in https://github.com/python/cpython/issues/123424 to deduplicate. It might be worthwhile awaiting a fix before submitting this one, to avoid conflicts and additional duplication.

Reply:
Thank you for the reminder. This PR currently has some testing failures, and I need to make further modifications today. Therefore, it is currently in draft status. We can review #123424 first

Reply:
Now that PR has been merged, the duplication is gone, which should simplify this PR. Thanks for your patience and sorry for the delay.

Reply:
it doesn't matter. I noticed that the upstream commit has made many changes that conflict with this PR, such as moving some code around. I'm not very good at resolving conflicts and might need a few days to study it. At the latest, I'll have it done by 2025/01/02